### ISSUE TITLE
collections.go:233: [PROGRAMMER ERROR] WARNING: Collection.hostName is empty!

### ISSUE THREAD

**Issue by schu** [writeas/writefreely#216]
```
ERROR: 2019/11/30 09:33:27 collections.go:233: [PROGRAMMER ERROR] WARNING: Collection.hostName is empty! Federation and many other things will fail! If you're seeing this in the wild, please report this bug and let us know what you were doing just before this: https://github.com/writeas/writefreely/issues/new?template=bug_report.md
ERROR: [repeated a few times]
```

Error message popped up upon first follower from a Mastodon instance. Follow worked but posts can't be loaded it seems ("No toots here!").

New single-user writefreely instance, version `v0.11.1`.

**Comment by thebaer** [comment]
Thanks for the report! Do you have the exact URL that was requested when this error came up? (It's usually the line directly following this one.) Or could you share more of your logs? That'll help us fix this.

**Comment by schu** [comment]
Opening a post for editing seems to trigger the error:

```
Dec 01 11:58:59 [snip] writefreely[1397]: ERROR: 2019/12/01 11:58:59 collections.go:233: [PROGRAMMER ERROR] WARNING: Collection.hostName is empty! Federation and many other things will fail! If you're seeing this in the wild, please report this bug and let us know what you were doing just before this: https://github.com/writeas/writefreely/issues/new?template=bug_report.md
Dec 01 11:58:59 [snip] writefreely[1397]: 2019/12/01 11:58:59 "GET /...snip.../edit" 200 2.679008ms "Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/76.0.3809.100 Safari/537.36"
```

From Mastodon I see "4 Toots" (correct number) but only three posts (first missing).

**Comment by thebaer** [comment]
Thanks. The discrepancy in Mastodon is likely something on their end -- e.g. Mastodon doesn't pull in old posts from _before_ you follow a blog, and that number is never guaranteed to match up to actual number of posts on a remote account.

But I'm taking a look at this log message now.

**Comment by grinapo** [comment]
Dunno if this was supposed to be fixed in v0.12.0 but it's not. 
Created a new site, new blog, tried to follow it then sent a direct message to it. :-) Okay, maybe that's over the top, but the error is there anyway.
